support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -362,7 +362,11 @@ private static void finish(@CheckForNull String cookie) {
                     return;
                 }
                 final AsynchronousExecution execution = runningTask.execution;
-                assert execution != null && runningTask.launcher != null;
+                if (execution == null) {
```

Comment:
Its body task _has_ been scheduled. Creating the `AsynchronousExecution` happens slightly later. If the body is very fast (typically just because there are no external tasks, just a `println`), then the body can finish in between the two. Definitely a corner case.